lib/auth/default_current_user_provider.rb

```diff
@@ -222,9 +238,15 @@ def log_on_user(user, session, cookies, opts = {})
     @env[CURRENT_USER_KEY] = user
   end
 
-  def cookie_hash(unhashed_auth_token)
+  def cookie_hash(unhashed_auth_token, user)
+    cookie = DiscourseAuthCookie.new(
+      token: unhashed_auth_token,
+      user_id: user.id,
+      trust_level: user.trust_level,
+      valid_till: UserAuthToken::ROTATE_TIME.from_now
```

Comment:
Using the rotate time is good here. However, ROTATE_TIME is the **minimumI** amount of time before the server will rotate a client's token.

Therefore, I think we should add a little leeway (maybe 1 minute or so), to make sure the client has been given a chance to rotate the token before we remove their rate-limit-bypassing ability.

Reply:
Done!